supervision/tracker/ByteTrack/matching.py

```diff
@@ -0,0 +1,201 @@
+import lap, cv2, time, scipy
+import numpy as np
+from scipy.spatial.distance import cdist
+from supervision.tracker.ByteTrack import kalman_filter
+
+def merge_matches(m1, m2, shape):
+    O,P,Q = shape
+    m1 = np.asarray(m1)
+    m2 = np.asarray(m2)
+
+    M1 = scipy.sparse.coo_matrix((np.ones(len(m1)), (m1[:, 0], m1[:, 1])), shape=(O, P))
+    M2 = scipy.sparse.coo_matrix((np.ones(len(m2)), (m2[:, 0], m2[:, 1])), shape=(P, Q))
+
+    mask = M1*M2
+    match = mask.nonzero()
+    match = list(zip(match[0], match[1]))
+    unmatched_O = tuple(set(range(O)) - set([i for i, j in match]))
+    unmatched_Q = tuple(set(range(Q)) - set([j for i, j in match]))
+
+    return match, unmatched_O, unmatched_Q
+
+
+def _indices_to_matches(cost_matrix, indices, thresh):
+    matched_cost = cost_matrix[tuple(zip(*indices))]
+    matched_mask = (matched_cost <= thresh)
+
+    matches = indices[matched_mask]
+    unmatched_a = tuple(set(range(cost_matrix.shape[0])) - set(matches[:, 0]))
+    unmatched_b = tuple(set(range(cost_matrix.shape[1])) - set(matches[:, 1]))
+
+    return matches, unmatched_a, unmatched_b
+
+
+def linear_assignment(cost_matrix, thresh):
+    if cost_matrix.size == 0:
+        return np.empty((0, 2), dtype=int), tuple(range(cost_matrix.shape[0])), tuple(range(cost_matrix.shape[1]))
+    matches, unmatched_a, unmatched_b = [], [], []
+    cost, x, y = lap.lapjv(cost_matrix, extend_cost=True, cost_limit=thresh)
```

Comment:
If possible, I want to reimplement `lap.lapjv` so we wouldn't need to drag the whole `lap` library as a dependency. ChatGPT claims `from scipy.optimize import linear_sum_assignment` is equivalent to `lap.lapjv`. We would need to take a look at it and confirm.

Reply:
I could not get successful results. Will check it in future.

Reply:
@SkalskiP @dbroboflow `lap.lapjv` and `from scipy.optimize import linear_sum_assignment` are equivalent. It is used for one to one match between trajectories and new detections.